File: Moeller Eaton V250/Doku/easyCOM_API.java
/***************************************************************************
 *
 *  easy_COM-DLL V2.5.0
 *  Interface declaration file for Java
 *
 *  Copyright (c) 2011  by Eaton Industries GmbH, Bonn, Germany.
 *  All rights reserved.
 *  Subject to alterations without notice.
 *
 *  All brand and product names are trademarks or registered trademarks of
 *  the owner concerned.
 *
 *  See AM_EASY_COM_G.PDF Document for further information.
 *
 ****************************************************************************
 */
 
import com.sun.jna.*;
import com.sun.jna.ptr.*;
 
/**
 * @brief "easyCOM Single connection API" using Java Native Access
 */
public class easyCOM_API
{
	static { Native.register("EASY_COM"); }

	public static native int Open_ComPort
		(
			byte com_port_nr,
			int baudrate
		);
		
	public static native int Close_ComPort
		(
		);

	public static native int Open_EthernetPort
		(
			String         szIpAddress,
			int            IpPort,
			int            baudrate,
			byte           no_baudrate_scan
		);

	public static native int Close_EthernetPort
		(
		);

	public static native int Set_UserWaitingTime
		(
			int            timeout_delay
		);

	public static native int Get_UserWaitingTime
		(
			IntByReference timeout_delay
		);

	public static native int GetCurrent_Baudrate
		(
			IntByReference baudrate
		);


	public static native int Start_Program
		(
			byte            net_id,
			ByteByReference errorcode
		);

	public static native int Stop_Program
		(
			byte            net_id,
			ByteByReference errorcode
		);
	

	public static native int Read_Clock
		(
			byte            net_id,
			ByteByReference year,
			ByteByReference month,
			ByteByReference day,
			ByteByReference hour,
			ByteByReference min
		);

	public static native int Write_Clock
		(
			byte            net_id,
			ByteByReference year,
			ByteByReference month,
			ByteByReference day,
			ByteByReference hour,
			ByteByReference min
		);


	public static native int Read_Object_Value
		(
			byte            net_id,
			byte            object,
			short           index,
			byte[]          data
		);

	public static native int Write_Object_Value
		(
			byte            net_id,
			byte            object,
			byte            index,
			short           length,
			byte[]          data
		);

	public static native int Read_Channel_YearTimeSwitch
		(
			byte            net_id,
			byte            index,
			byte            channel,
			ByteByReference on_year,
			ByteByReference on_month,
			ByteByReference on_day,
			ByteByReference off_year,
			ByteByReference off_month,
			ByteByReference off_day
		);

	public static native int Write_Channel_YearTimeSwitch
		(
			byte            net_id,
			byte            index,
			byte            channel,
			ByteByReference on_year,
			ByteByReference on_month,
			ByteByReference on_day,
			ByteByReference off_year,
			ByteByReference off_month,
			ByteByReference off_day
		);

	public static native int Read_Channel_7DayTimeSwitch
		(
			byte            net_id,
			byte            index,
			byte            channel,
			ByteByReference DY1,
			ByteByReference DY2,
			ByteByReference on_hour,
			ByteByReference on_minute,
			ByteByReference off_hour,
			ByteByReference off_minute
		);

	public static native int Write_Channel_7DayTimeSwitch
		(
			byte            net_id,
			byte            index,
			byte            channel,
			ByteByReference DY1,
			ByteByReference DY2,
			ByteByReference on_hour,
			ByteByReference on_minute,
			ByteByReference off_hour,
			ByteByReference off_minute
		);


	public static native int Unlock_Device
		(
			byte            net_id,
			String          szPassword,
			ByteByReference errorcode
		);

	public static native int Lock_Device
		(
			byte            net_id,
			ByteByReference errorcode
		);

	public static native int GetLastSystemError
		(
		);

		
	// usage demo
	public static void main(String[] args)
	{
		int success;
		IntByReference pWord = new IntByReference( 0 );

		System.out.println( "Hallo easy:" );
		
		success = Set_UserWaitingTime( 500 );
		System.out.println( "Set_UserWaitingTime( 500 ) => " + success );
		
		success = Get_UserWaitingTime( pWord );
		System.out.println( "Get_UserWaitingTime( ) => " + pWord.getValue() );

		success = Open_ComPort( (byte)1 , 19200 );
		System.out.println( "OpenComPort( 1, 19200 ) => " + success );

		if ( success == 0 )
		{
			byte NT0 = 0;
			byte bData[] = new byte[80];

			ByteByReference pError = new ByteByReference( (byte)0 );
			ByteByReference pYY = new ByteByReference( (byte)0 );
			ByteByReference pMM = new ByteByReference( (byte)0 );
			ByteByReference pDD = new ByteByReference( (byte)0 );
			ByteByReference pHH = new ByteByReference( (byte)0 );
			ByteByReference pMIN = new ByteByReference( (byte)0 );

			success = GetCurrent_Baudrate( pWord );
			if ( success == 0 )
				System.out.println( "GetCurrent_Baudrate( ) => " + pWord.getValue() );

			success = Read_Clock( NT0, pYY, pMM, pDD, pHH, pMIN );
			if ( success == 0 )
			{
				java.util.GregorianCalendar dt = new java.util.GregorianCalendar();
				java.text.DateFormat dtFormatter = java.text.SimpleDateFormat.getDateTimeInstance( java.text.SimpleDateFormat.MEDIUM, java.text.SimpleDateFormat.SHORT );
				
				dt.set( 2000+pYY.getValue(), pMM.getValue()-1, pDD.getValue(), pHH.getValue(), pMIN.getValue() );
				System.out.println( "Read_Clock(...) => " + dtFormatter.format( dt.getTime() ) );
			}
			else
			{
				System.out.println( "Read_Clock(...) => error " + success );
			}

			success = Read_Object_Value( NT0, (byte)0, (short)0, bData );
			if ( success == 0 )
			{
				System.out.printf( "MC_Read_Object_Value(I) => hex %02X%02X \n", bData[1], bData[0] );
			}
			else
			{
				System.out.println( "Read_Object_Value(I) => error " + success );
			}

		
			success = Stop_Program( NT0, pError );
			System.out.println( "Stop_Program(...) => " + success );

			success = Start_Program( NT0, pError );
			System.out.println( "Start_Program(...) => " + success );
		
			success = Lock_Device( NT0, pError );
			if ( success != 0 )
			System.out.println( "Lock_Device(...) => " + success );

			success = Unlock_Device( NT0, "123456", pError );
			System.out.println( "Unlock_Device(...) => " + success );
			
			success = Close_ComPort( );
			System.out.println( "Close_ComPort( ) => " + success );
		}
		
		success = Open_EthernetPort( "192.168.1.162", 10001, 38400, (byte)0 );
		System.out.println( "Open_EthernetPort(...) => " + success );
			
		success = Close_EthernetPort();
		System.out.println( "Close_EthernetPort(...) => " + success );
	}
}

File: Moeller Eaton V250/Doku/easyComApi.vb
'***************************************************************************
' *
' *  easy_COM-DLL V2.5.0
' *  Interface declaration file for Microsoft Visual Basic 2005/2008 (VB.NET)
' *
' *  Copyright (c) 2011 by Eaton Industries GmbH, Bonn, Germany.
' *  All rights reserved.
' *  Subject to alterations without notice.
' *
' *  All brand and product names are trademarks or registered trademarks of the owner concerned.
' *
' *  See AM_EASY_COM_G.PDF Document for further information.
' *
'****************************************************************************
Public Class easyCOM

	'****************************************************************************
	' public interface functions for single connection mode

	Declare Ansi Function Open_ComPort Lib "EASY_COM.DLL" _
	(ByVal com_port_nr As Byte, _
	 ByVal baudrate As Integer) As Integer

	Declare Ansi Function Close_ComPort Lib "EASY_COM.DLL" () As Integer

	Declare Ansi Function GetCurrent_Baudrate Lib "EASY_COM.DLL" _
	(ByRef baudrate As Integer) As Integer

	Declare Ansi Function Set_UserWaitingTime Lib "EASY_COM.DLL" _
	(ByVal waittime As Integer) As Integer

	Declare Ansi Function Get_UserWaitingTime Lib "EASY_COM.DLL" _
	(ByRef waittime As Integer) As Integer

	Declare Ansi Function Open_EthernetPort Lib "EASY_COM.DLL" _
	(ByVal szIpAddress As String, _
	 ByVal IpPort As Integer, _
	 ByVal baudrate As Integer, _
	 ByVal no_baudrate_scan As Boolean) As Integer

	Declare Ansi Function Close_EthernetPort Lib "EASY_COM.DLL" () As Integer

	Declare Ansi Function Start_Program Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, ByRef Errorcode As Byte) As Integer

	Declare Ansi Function Stop_Program Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, ByRef Errorcode As Byte) As Integer

	Declare Ansi Function Read_Clock Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByRef year As Byte, _
	 ByRef month As Byte, _
	 ByRef day As Byte, _
	 ByRef hour As Byte, _
	 ByRef min As Byte) As Integer

	Declare Ansi Function Write_Clock Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByRef year As Byte, _
	 ByRef month As Byte, _
	 ByRef day As Byte, _
	 ByRef hour As Byte, _
	 ByRef min As Byte) As Integer

	Declare Ansi Function Read_Object_Value Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal obj As Byte, _
	 ByVal index As UShort, _
	 ByRef data As Byte) As Integer

	Declare Ansi Function Write_Object_Value Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal obj As Byte, _
	 ByVal index As UShort, _
	 ByVal length As Byte, _
	 ByRef data As Byte) As Integer

	Declare Ansi Function Read_Channel_YearTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef on_year As Byte, _
	 ByRef on_month As Byte, _
	 ByRef on_day As Byte, _
	 ByRef off_year As Byte, _
	 ByRef off_month As Byte, _
	 ByRef off_day As Byte) As Integer

	Declare Ansi Function Write_Channel_YearTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef on_year As Byte, _
	 ByRef on_month As Byte, _
	 ByRef on_day As Byte, _
	 ByRef off_year As Byte, _
	 ByRef off_month As Byte, _
	 ByRef off_day As Byte) As Integer

	Declare Ansi Function Read_Channel_7DayTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef DY1 As Byte, _
	 ByRef DY2 As Byte, _
	 ByRef on_hour As Byte, _
	 ByRef on_minute As Byte, _
	 ByRef off_hour As Byte, _
	 ByRef off_minute As Byte) As Integer

	Declare Ansi Function Write_Channel_7DayTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef DY1 As Byte, _
	 ByRef DY2 As Byte, _
	 ByRef on_hour As Byte, _
	 ByRef on_minute As Byte, _
	 ByRef off_hour As Byte, _
	 ByRef off_minute As Byte) As Integer

	Declare Ansi Function Unlock_Device Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByVal szPassword As String, _
	 ByRef Errorcode As Byte) As Integer

	Declare Ansi Function Lock_Device Lib "EASY_COM.DLL" _
	(ByVal net_id As Byte, _
	 ByRef Errorcode As Byte) As Integer

	'****************************************************************************
	' public interface functions for multiple connection mode
	'
	Declare Ansi Function MC_Open_ComPort Lib "EASY_COM.DLL" _
	(ByRef handle As Integer, _
	 ByVal com_port_nr As Byte, _
	 ByVal baudrate As Integer) As Integer

	Declare Ansi Function MC_Close_ComPort Lib "EASY_COM.DLL" _
	(ByVal handle As Integer) As Integer

	Declare Ansi Function MC_GetCurrent_Baudrate Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByRef baudrate As Integer) As Integer

	Declare Ansi Function MC_Set_UserWaitingTime Lib "EASY_COM.DLL" _
	(ByVal waittime As Integer) As Integer

	Declare Ansi Function MC_Get_UserWaitingTime Lib "EASY_COM.DLL" _
	(ByRef waittime As Integer) As Integer

	Declare Ansi Function MC_Open_EthernetPort Lib "EASY_COM.DLL" _
	(ByRef handle As Integer, _
	 ByVal szIpAddress As String, _
	 ByVal IpPort As Integer, _
	 ByVal baudrate As Integer, _
	 ByVal no_baudrate_scan As Boolean) As Integer

	Declare Ansi Function MC_Close_EthernetPort Lib "EASY_COM.DLL" _
	(ByVal handle As Integer) As Integer

	Declare Ansi Function MC_CloseAll Lib "EASY_COM.DLL" () As Integer

	Declare Ansi Function MC_Start_Program Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, ByRef Errorcode As Byte) As Integer

	Declare Ansi Function MC_Stop_Program Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, ByRef Errorcode As Byte) As Integer

	Declare Ansi Function MC_Read_Clock Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByRef year As Byte, _
	 ByRef month As Byte, _
	 ByRef day As Byte, _
	 ByRef hour As Byte, _
	 ByRef min As Byte) As Integer

	Declare Ansi Function MC_Write_Clock Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByRef year As Byte, _
	 ByRef month As Byte, _
	 ByRef day As Byte, _
	 ByRef hour As Byte, _
	 ByRef min As Byte) As Integer

	Declare Ansi Function MC_Read_Object_Value Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal obj As Byte, _
	 ByVal index As UShort, _
	 ByVal length As Byte, _
	 ByRef data As Byte) As Integer


	Declare Ansi Function MC_Write_Object_Value Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal obj As Byte, _
	 ByVal index As UShort, _
	 ByVal length As Byte, _
	 ByRef data As Byte) As Integer

	Declare Ansi Function MC_Read_Channel_YearTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef on_year As Byte, _
	 ByRef on_month As Byte, _
	 ByRef on_day As Byte, _
	 ByRef off_year As Byte, _
	 ByRef off_month As Byte, _
	 ByRef off_day As Byte) As Integer

	Declare Ansi Function MC_Write_Channel_YearTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef on_year As Byte, _
	 ByRef on_month As Byte, _
	 ByRef on_day As Byte, _
	 ByRef off_year As Byte, _
	 ByRef off_month As Byte, _
	 ByRef off_day As Byte) As Integer

	Declare Ansi Function MC_Read_Channel_7DayTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef DY1 As Byte, _
	 ByRef DY2 As Byte, _
	 ByRef on_hour As Byte, _
	 ByRef on_minute As Byte, _
	 ByRef off_hour As Byte, _
	 ByRef off_minute As Byte) As Integer

	Declare Ansi Function MC_Write_Channel_7DayTimeSwitch Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal index As Byte, _
	 ByVal channel As Byte, _
	 ByRef DY1 As Byte, _
	 ByRef DY2 As Byte, _
	 ByRef on_hour As Byte, _
	 ByRef on_minute As Byte, _
	 ByRef off_hour As Byte, _
	 ByRef off_minute As Byte) As Integer

	Declare Ansi Function MC_Unlock_Device Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByVal szPassword As String, _
	 ByRef Errorcode As Byte) As Integer

	Declare Ansi Function MC_Lock_Device Lib "EASY_COM.DLL" _
	(ByVal handle As Integer, _
	 ByVal net_id As Byte, _
	 ByRef Errorcode As Byte) As Integer

	'****************************************************************************
	' Windows system access functions

	Declare Ansi Function GetLastSystemError Lib "EASY_COM.DLL" () As Integer

	'****************************************************************************
	' possible function return codes

	Public Const easy_COM_Err_Success As Integer = 0

	Public Const easy_COM_Err_InvalidParameter As Integer = 1

	Public Const easy_COM_Err_DeviceAdverseResponse As Integer = 2
	Public Const easy_COM_Err_DeviceNoResponse As Integer = 3
	Public Const easy_COM_Err_DevicePasswortLock As Integer = 4
	Public Const easy_COM_Err_DeviceTypeUnknown As Integer = 5

	Public Const easy_COM_Err_WindowsSystemError As Integer = 6

	Public Const easy_COM_Err_ComPortNotOpen As Integer = 7
	Public Const easy_COM_Err_ComPortNotExist As Integer = 8
	Public Const easy_COM_Err_ComPortCannotAccessed As Integer = 9
	Public Const easy_COM_Err_ComGeneralError As Integer = 10

	Public Const easy_COM_Err_InternalError As Integer = 11

	Public Const easy_COM_Err_FB_NotExist As Integer = 12
	Public Const easy_COM_Err_FB_NoConstParameter As Integer = 13

	Public Const easy_COM_Err_TcpPortNotExist As Integer = 14
	Public Const easy_COM_Err_TcpPortNoBaudrateScanPossible As Integer = 15

	Public Const easy_COM_Err_HandleInvalid As Integer = 16

End Class

